src/main/java/com/google/sps/servlets/TaskServlet.java

```diff
@@ -62,15 +136,22 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     long creationTime = System.currentTimeMillis();
 
+    UserService userService = UserServiceFactory.getUserService();
+    String userId = userService.getCurrentUser().getUserId();
+
     // Create an Entity that stores the input comment
     Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("userId", userId);
     taskEntity.setProperty("detail", taskDetail);
     taskEntity.setProperty("timestamp", creationTime);
     taskEntity.setProperty("reward", rewardPts);
     taskEntity.setProperty("status", "OPEN");
     taskEntity.setProperty("Owner", "Leonard");
     taskEntity.setProperty("Helper", "N/A");
     taskEntity.setProperty("Address", "4xxx Cxxxxx Avenue, Pittsburgh, PA 15xxx");
+    taskEntity.setProperty("zipcode", "59715");
+    taskEntity.setProperty("country", "United States");
```

Comment:
Will we be able to decide the zipcode and country automatically with Maps API? If not, or if it is really hard, I can add two more fields that let user input their country and zipcode.

Reply:
I don't think it should be too difficult, but that's not a bad idea to start with - that way we can stop using hardcoded data and once we get around it we can change it so that it retrieves the zipcode and country from the address. That would be great, thanks!